### PR TITLE
Ci kind install

### PR DESCRIPTION
**Pull request by marcofranssen**
This PR resolves some issues when the spire chart is installed in a namespace named different than `spire`.
The chart install also failed when a different release name was used than `spire`.

This PR uses a kind cluster to install and test the chart with a random namespace and release.

### CODE REVIEW

File: charts/spire/templates/tests/test-connection.yaml [hunk 1]
```diff
@@ -1,15 +0,0 @@
-apiVersion: v1
```
Comment (gjkamstra) [inline charts/spire/templates/tests/test-connection.yaml:1]: Why did you delete this file?

**Review by gjkamstra — APPROVED** [review]
Looks good. Nicely shows the value of CI.
I am just wondering why you removed the test-connection file.

File: charts/spire/templates/tests/test-connection.yaml (hunk 1 above)
Comment (marcofranssen) [reply charts/spire/templates/tests/test-connection.yaml:1]: WGET doesn't work on a grpc service. This caused the CI to fail as it will also run the tests. We would have to create a new test-connection deployment based on a grpc_cli to test the grpc endpoints. For that we need a trustworthy docker image. Considering building one for this purpose.

Comment (gjkamstra) [reply charts/spire/templates/tests/test-connection.yaml:1]: Clear, thanks for clarifying.
